### PR TITLE
Add Accessiblity Features to the Contribution Screen

### PR DESCRIPTION
**Pull request by SudharakaP**
This PR provides the following basic features. I've tested this with [Orca](https://help.gnome.org/users/orca/stable/)

- Add group details to buttons
- Announce pressed state of the buttons
- Announce the message dialog
- Auto focus on the cancel button of the dialog
- Announce the percentage selection at the bottom of the form (the donation select input)

The following needs to be still completed.

~~- Ideally trap the focus within the dialog after it opens (i.e: we need to make sure the tabs don't flow out of the dialog; they should wrap)~~
~~- Need a better solution to the donation select input as we seem to be using react-select and that seems to have a lot of accessibility issues; https://github.com/JedWatson/react-select/issues?q=is%3Aissue+is%3Aopen+accessibility~~

Resolves https://github.com/opencollective/opencollective/issues/3790

### CODE REVIEW

**Comment by SudharakaP** [comment]
@Betree : This is ready for review. I have removed the formatting of the drop-down label to make it simpler so that it works with screen readers easily. Please take a look and let me know. The only additional thing remaining to do here is probably try to trap the focus within the dialog when it opens. I will try to do it in the meantime. 🤔

**Comment by SudharakaP** [comment]
@Betree : Added focus trap as well, using the [`focus-trap-react` library](https://github.com/focus-trap/focus-trap-react). So I think this is fully done. Let me know if there's anything else I might have missed. It's a bit hard to think what else is needed here as I don't use screen readers on a day to day basis; maybe we need help from someone who does? 🤔

File: components/ConfirmationModal.js [hunk 1]
```diff
@@ -52,40 +53,43 @@ const ConfirmationModal = ({
...
-              } catch (e) {
-                setSubmitting(false);
-                throw e;
-              }
-            }}
-          >
-            {continueLabel || formatMessage(confirmBtnMsgs[type])}
-          </StyledButton>
-        </Container>
+        <FocusTrap focusTrapOptions={{ clickOutsideDeactivates: true }}>
```
Comment (Betree) [inline components/ConfirmationModal.js:60]: Shouldn't we put the focus trap directly on `StyledModal`? As it sometimes hold forms, having that as a global pattern seems like a good improvement

File: components/contribution-flow/FeesOnTopInput.js [hunk 2]
```diff
@@ -41,11 +36,11 @@ const getOptions = (amount, currency) => {
       return getOptionFromPercentage(amount, currency, percentage);
     }),
     {
-      label: <FormattedMessage id="NoThankYou" defaultMessage="No, thank you" />,
```
Comment (Betree) [inline components/contribution-flow/FeesOnTopInput.js:44]: We should keep these internationalized. If the issue is that we want to have them as strings, we should use a `defineMessages` + `useIntl()` + `formatMessage`.

File: components/contribution-flow/FeesOnTopInput.js [hunk 3]
```diff
@@ -27,11 +26,7 @@ const getOptionFromPercentage = (amount, currency, percentage) => {
     // Value must be unique, so we set a special key if feeAmount is 0
     value: feeAmount || `${percentage}%`,
     percentage,
-    label: (
-      <span>
-        {formatCurrency(feeAmount, currency)} <Span color="black.500">({percentage * 100}%)</Span>
```
Comment (Betree) [inline components/contribution-flow/FeesOnTopInput.js:32]: I assume that a string was needed as an output, if that's the case could we rather put that as `aria-title` (or similar, I'm not sure which one to use) to keep the formatting?

Also relying on the standard `formatCurrency` is a good practice, would be great if we can keep it

**Review by Betree — COMMENTED** [review]
Just tried with my OS screen reader and the result is great, good job!

The trap for the modal also seems to do its job, I wonder if we can extend that to all modals (see below)?

File: components/ConfirmationModal.js (hunk 1 above)
Comment (SudharakaP) [reply components/ConfirmationModal.js]: @Betree : Good idea. Done. 😄

File: components/contribution-flow/FeesOnTopInput.js (hunk 3 above)
Comment (SudharakaP) [reply components/contribution-flow/FeesOnTopInput.js:32]: Done. 👍🏽

File: components/contribution-flow/FeesOnTopInput.js (hunk 2 above)
Comment (SudharakaP) [reply components/contribution-flow/FeesOnTopInput.js:44]: @Betree : Didn't know about this, sorry. Adjusted the code. Now it should be good. 😄

**Comment by SudharakaP** [comment]
@Betree : Sorry, I think there is a better way of handling the options array with the use of `formatOptionLabel` which I've used now. Please take a look and let me know if you see any issue. The formatting should not change with this method; it should remain as it is. 😄

**Review by Betree — APPROVED** [review]

**Pull request merged**